### PR TITLE
docs: revamp entire Jan guides

### PR DESCRIPTION
**Pull request by hieu-jan**
## Add Quickstart Docs and implement Custom plugin

- Add the quickstart docs and configure the sidebar and `docusaurus.config.js` file.
- Add the Docusaurus custom Plugin to fetch recent data from GitHub Releases page.
- Add another `.env.example` file variable that store GitHub access token

## Checklist before merging:
- [ ] Resolve the issue of Changelog
- [ ] Re-allocate the image files under the assets folder of each page 
- [ ] Resolve current incorrect image source, incorrect redirect, typo in documentation

## Fixes Issues

- Closes #
- Closes #

## Self Checklist

- [ ] Added relevant comments, esp in complex areas
- [x] Updated docs (for bug fixes / features)
- [ ] Created issues for follow-up changes or refactoring needed

### CODE REVIEW

File: docs/docs/quickstart/quickstart.mdx [hunk 1]
```diff
@@ -0,0 +1,50 @@
+---
+sidebar_position: 1
+hide_table_of_contents: true
+---
+
+import installImageURL from '../../static/img/homepage-new/jan-ai-quickstart.png';
```
Comment (hieu-jan) [inline docs/docs/quickstart/quickstart.mdx:6]: Is it a redundant import as import but no use `installImageURL`. Personally, I think it's better for create assets folder inside quickstart folder for these images.

File: docs/docs/quickstart/quickstart.mdx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+3. Go to the **Hub** under the **Thread** section and select the AI model that you want to use. For more info, go to the [Using Models](category/using-models) section.
+
+4. A new thread will be added. You can use Jan in the thread with the AI model that you selected before. */}
+
+### Step 1: Install Jan
+
+Go to [Jan.ai](https://jan.ai/) > Select your operating system > Install the program.
+
+To learn more about system requirements for your operating system, go to [Installation guide](/docs/install).
```
Comment (hieu-jan) [inline docs/docs/quickstart/quickstart.mdx:36]: Error link in preview mode

**Comment by 0xSage** [comment]
Gentle reminder, can we: 
1. ship smaller PRs
2. use conventional commits: https://www.conventionalcommits.org/en/v1.0.0/#summary
cc @hieu-jan @aindrajaya @cahyosubroto

**Comment by 0xSage** [comment]
Can I clarify: what's the difference btw `quickstart` and `user guide`?

**Review by aindrajaya — COMMENTED** [review]
done, already reviewed. Thanks @hieu-jan

**Comment by hieu-jan** [comment]
> done, already reviewed. Thanks @hieu-jan

Hi @aindrajaya, seems that you're already added to the member of repo. Could you make the changes based on feedback directly to this branch. Also for the future pull request, could you create a pull request from a branch inside repo to get docs preview, so I won't need to re-create another pull request for pointing to get preview like this one.

**Comment by 0xSage** [comment]
@hieu-jan to work with pena team and just get this one merged.
Delete & remove a lot of the finicky changes.

File: docs/docs/quickstart/quickstart.mdx (hunk 2 above)
Comment (aindrajaya) [reply docs/docs/quickstart/quickstart.mdx:36]: Already updated.

File: docs/docs/quickstart/quickstart.mdx (hunk 1 above)
Comment (aindrajaya) [reply docs/docs/quickstart/quickstart.mdx]: Already updated by adding the `./assets` folder inside the quickstart.

File: docs/docs/quickstart/models/import-models.mdx [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+### 2. Drag & Drop the Model
+
+Drag and drop your model binary into this folder, ensuring the `modelname.gguf` is the same name as the folder name, e.g. `models/modelname`.
+
+### 3. Done!
+
+If your model doesn't show up in the **Model Selector** in conversations, **restart the app** or contact us via our [Discord community](https://discord.gg/Dt7MxDyNNZ).
+
+## Older versions - before v0.44
```
[the remark this answers is not in the document]
Comment (aindrajaya) [reply docs/docs/quickstart/models/import-models.mdx]: Already updated.

File: docs/docs/quickstart/quickstart.mdx (hunk 2 above)
Comment (hieu-jan) [reply docs/docs/quickstart/quickstart.mdx:36]: Could you re-check the newest preview. I think this issue still persists 
https://b3bbeeb2.jan-qn8.pages.dev/quickstart/

File: docs/docs/quickstart/quickstart.mdx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+### Step 2: Select AI Model
+
+Before using Jan, you need to select an AI model that based on your hardware capabilities and specifications.
+
+Each model has their purposes, capabilities, and different requirements.
+
+To select AI models: Go to the **Hub** > select the models that you would like to install.
+
+For more info, go to [list of supported models](/docs/models-list/).
```
[the remark this answers is not in the document]
Comment (hieu-jan) [reply docs/docs/quickstart/quickstart.mdx:46]: Could you re-check the newest preview. I think this issue still persists
https://b3bbeeb2.jan-qn8.pages.dev/quickstart/

File: docs/docs/quickstart/integration/azure.mdx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+```json
+"parameters":{
+  "temperature": 0.7,
+  "top_p": 0.95,
+  "stream": true,
+  "max_tokens": 4096,
+  "frequency_penalty": 0,
+  "presence_penalty": 0
+}
+```
```
Comment (hieu-jan) [inline docs/docs/quickstart/integration/azure.mdx:84]: As we use the model via Azure OpenAI, adding this additional part is not fit with current content. 

I think we can add a section guiding user on how to customize the parameters only and point to this link 

https://learn.microsoft.com/en-us/azure/ai-services/openai/reference

File: docs/docs/quickstart/integration/vscode.mdx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+```json title="~/.continue/config.json"
+{
+  "models": [
+    {
+      "title": "Jan",
+      "provider": "openai",
+      "model": "mistral-ins-7b-q4",
+      "apiKey": "EMPTY",
+      "apiBase": "http://localhost:1337"
```
Comment (hieu-jan) [inline docs/docs/quickstart/integration/vscode.mdx:61]: "apiBase": "http://localhost:1337/v1"

File: docs/docs/quickstart/integration/vscode.mdx [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    }
+  ]
+}
+```
+2. Ensure the file has the following configurations:
+    - Ensure `openai` is selected as the `provider`.
+    - Match the `model` with the one enabled in the Jan API Server.
+    - Set `apiBase` to `http://localhost:1337`.
```
Comment (hieu-jan) [inline docs/docs/quickstart/integration/vscode.mdx:69]: http://localhost:1337/v1

https://dev.jan.ai/guides/integrations/continue/

File: docs/docs/quickstart/models/customize-engine.mdx [hunk 8]
```diff
@@ -0,0 +1,187 @@
+---
+sidebar_position: 3
```
Comment (hieu-jan) [inline docs/docs/quickstart/models/customize-engine.mdx:2]: sidebar_position seems incorrect. 

Currently having two manual import 
image

File: docs/docs/quickstart/integration/azure.mdx (hunk 5 above)
Comment (hieu-jan) [reply docs/docs/quickstart/integration/azure.mdx:84]: For Azure docs, I think we might need to maintain these screenshots so user may be easily setting up without confused. As current screenshot already point to the main steps in setting up on Azure, or important parts when users modify their own model.json

File: docs/docs/quickstart/integration/discord.mdx [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+| LOCAL_SERVER_URL | URL of your local API server for LLMs starting with `local/`. (Default: `http://localhost:5000/v1`) |
+| LOCAL_API_KEY | API key for your local API server with LLMs starting with `local/`. Usually safe to leave blank. |
+
+### Step 4: Invite the bot to your Discord server using this URL (replace `CLIENT_ID` with your Discord application's client ID from the OAuth2 tab)
+
+```
+https://discord.com/api/oauth2/authorize?client_id=(CLIENT_ID)&permissions=412317273088&scope=bot
+```
+
+### STep 5: Run the bot
```
Comment (hieu-jan) [inline docs/docs/quickstart/integration/discord.mdx:54]: Typo

File: docs/docs/quickstart/integration/discord.mdx [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+| CUSTOM_SYSTEM_PROMPT | Adjust the bot's behavior as needed. |
+| CUSTOM_DISCORD_STATUS | Set a custom message for the bot's Discord profile. (Max 128 characters) | 
+| ALLOWED_CHANNEL_IDS | Enter Discord channel IDs where the bot can send messages, separated by commas. Leave blank to allow all channels. |
+| ALLOWED_ROLE_IDS | Enter Discord role IDs allowed to use the bot, separated by commas. Leave blank to allow everyone. Including at least one role also disables DMs. |
+| MAX_IMAGES | Max number of image attachments allowed per message when using a vision model. (Default: `5`) |
+| MAX_MESSAGES | Max messages allowed in a reply chain. (Default: `20`) |
+| LOCAL_SERVER_URL | URL of your local API server for LLMs starting with `local/`. (Default: `http://localhost:5000/v1`) |
+| LOCAL_API_KEY | API key for your local API server with LLMs starting with `local/`. Usually safe to leave blank. |
+
+### Step 4: Invite the bot to your Discord server using this URL (replace `CLIENT_ID` with your Discord application's client ID from the OAuth2 tab)
```
Comment (hieu-jan) [inline docs/docs/quickstart/integration/discord.mdx:48]: Could we shorten the title to make it concise 
image

**Comment by hieu-jan** [comment]
Currently, some example file in Quickstart is removing highlight the import line as guide. Should we remain these syntax as current User Guide.

File: docs/docs/quickstart/integration/discord.mdx (hunk 9 above)
Comment (aindrajaya) [reply docs/docs/quickstart/integration/discord.mdx]: Already updated

File: docs/docs/quickstart/models/customize-engine.mdx (hunk 8 above)
Comment (aindrajaya) [reply docs/docs/quickstart/models/customize-engine.mdx]: It is already fixed.

File: docs/docs/quickstart/integration/vscode.mdx (hunk 7 above)
Comment (aindrajaya) [reply docs/docs/quickstart/integration/vscode.mdx]: Already fixed.

File: docs/docs/quickstart/integration/vscode.mdx (hunk 6 above)
Comment (aindrajaya) [reply docs/docs/quickstart/integration/vscode.mdx]: Already fixed.

File: docs/docs/quickstart/integration/azure.mdx (hunk 5 above)
Comment (aindrajaya) [reply docs/docs/quickstart/integration/azure.mdx]: Sorry, I forgot to place it in the wrong section. Yes, it shouldn't be there, and it's already removed. Thanks

**Pull request merged**